packages/x-components/src/x-modules/search/store/types.ts

```diff
@@ -106,7 +106,7 @@ export interface SearchMutations extends StatusMutations {
    *
    * @param origin - The new origin.
    */
-  setOrigin(origin: QueryOrigin | undefined): void;
+  setOrigin(origin: QueryOrigin | undefined | null): void;
```

Comment:
what's the difference between having `undefined` origin and having `null` origin? do we need both?

Reply:
They are equivalent, indeed they should be transformed both to `null`. The point is that the `origin` is optional in the metadata, and this mutation params were extended to support also that case.

Reply:
with the change I suggest below, the `createQueryOrigin` can always return an origin or `null`